### PR TITLE
Catch TypeLoadException during plugin load

### PR DESCRIPTION
**Pull request by Ullmie02**
**Changes**

**Issues**
Fixes server startup with incompatible plugins

### CODE REVIEW

File: Emby.Server.Implementations/Plugins/PluginManager.cs [hunk 1]
```diff
@@ -122,6 +122,12 @@ public IEnumerable<Assembly> LoadAssemblies()
                         ChangePluginState(plugin, PluginStatus.Malfunctioned);
                         continue;
                     }
+                    catch (TypeLoadException ex)
```
Comment (cvium) [inline Emby.Server.Implementations/Plugins/PluginManager.cs:125]: The point is to stop crashing, so in this case you'd want to catch the general Exception

File: Emby.Server.Implementations/Plugins/PluginManager.cs [hunk 2]
```diff
@@ -112,9 +112,16 @@ public IEnumerable<Assembly> LoadAssemblies()
...
+                        try
+                        {
+                            assembly.GetExportedTypes();
+                        }
+                        catch (TypeLoadException ex) // Undocumented exception
+                        {
+                            _logger.LogError(ex, "Failed to load assembly {Path}. This error occurs when a plugin references an incompatible version of one of the shared libraries. Disabling plugin.", file);
+                            ChangePluginState(plugin, PluginStatus.NotSupported);
+                            continue;
+                        }
```
Comment (Bond-009) [inline Emby.Server.Implementations/Plugins/PluginManager.cs:124]: This doesn't need to be nested in the other try-catch

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/Plugins/PluginManager.cs [hunk 3]
```diff
@@ -122,6 +120,12 @@ public IEnumerable<Assembly> LoadAssemblies()
                         ChangePluginState(plugin, PluginStatus.Malfunctioned);
                         continue;
                     }
+                    catch (TypeLoadException ex) // Undocumented exception
+                    {
+                        _logger.LogError(ex, "Failed to load assembly {Path}. This error occurs when a plugin references an incompatible version of one of the shared libraries. Disabling plugin.", file);
+                        ChangePluginState(plugin, PluginStatus.NotSupported);
+                        continue;
+                    }
```
Comment (cvium) [inline Emby.Server.Implementations/Plugins/PluginManager.cs:128]:
```suggestion
                    }
                    catch (Exception ex)
                    {
                        _logger.LogError(ex, "Failed to load assembly {Path}. Unknown exception was thrown. Disabling plugin.", file);
                        ChangePluginState(plugin, PluginStatus.Malfunctioned);
                        continue;
                    }
```

Comment (Bond-009) [reply Emby.Server.Implementations/Plugins/PluginManager.cs:128]: Analyzer will fail, need to disable the warning

**Review by cvium — APPROVED** [review]

**Review by Ladette19 — COMMENTED** [review]
Hello-World

**Pull request merged**